src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -250,7 +252,7 @@ internal void SendWorkerInitRequest(RpcEvent startEvent)
             _inboundWorkerEvents.Where(msg => msg.MessageType == MsgType.WorkerInitResponse)
                 .Timeout(workerInitTimeout)
                 .Take(1)
-                .Subscribe(PublishRpcChannelReadyEvent, HandleWorkerError);
+                .Subscribe(SendDownloadDependencyRequest, HandleWorkerError);
```

Comment:
New message SendDownloadDependencyRequest needs to follow PublishRpcChannelReadyEvent. Publishing Channel ready event means that the communication channel is established between host and the language worker and are ready to communicate

Reply:
@pragnagopa WorkerInitRequest is sent to language worker before PublishRpcChannelReadyEvent is published, and if the on the success language worker sends WorkerInitResponse, and looking at language worker code it seems the communication port for gRPC between host and worker is established by that time, hence I wanted to make sure before function load the dependency modules are downloaded. I can sync up with you on this.